ui/src/main/webapp/src/app/components/reports/source/source-report.component.ts

```diff
@@ -0,0 +1,283 @@
+import "prismjs/components/prism-scala";
+import "prismjs/components/prism-scheme";
+import "prismjs/components/prism-scss";
+import "prismjs/components/prism-smalltalk";
+import "prismjs/components/prism-smarty";
+import "prismjs/components/prism-sql";
+import "prismjs/components/prism-stylus";
+import "prismjs/components/prism-swift";
+import "prismjs/components/prism-tcl";
+import "prismjs/components/prism-textile";
+import "prismjs/components/prism-twig";
+import "prismjs/components/prism-typescript";
+import "prismjs/components/prism-verilog";
+import "prismjs/components/prism-vhdl";
+import "prismjs/components/prism-vim";
+import "prismjs/components/prism-wiki";
+import "prismjs/components/prism-yaml";
+
+// Broken at the moment (2016-12-05, jsightler)
+//import "prismjs/components/prism-crystal"
+//import "prismjs/components/prism-objectivec"
+
+import "prismjs/plugins/keep-markup/prism-keep-markup";
+
+import * as showdown from "showdown";
+
+import {FileModelService} from "../../../services/graph/file-model.service";
+import {FileModel} from "../../../generated/tsModels/FileModel";
+import {ClassificationService} from "../../../services/graph/classification.service";
+import {ClassificationModel} from "../../../generated/tsModels/ClassificationModel";
+import {InlineHintModel} from "../../../generated/tsModels/InlineHintModel";
+import {HintService} from "../../../services/graph/hint.service";
+import {LinkModel} from "../../../generated/tsModels/LinkModel";
+import {SourceFileModel} from "../../../generated/tsModels/SourceFileModel";
+import {GraphJSONToModelService} from "../../../services/graph/graph-json-to-model.service";
+import {Http} from "@angular/http";
+import {LinkableModel} from "../../../generated/tsModels/LinkableModel";
+import {Observable} from "rxjs";
+
+@Component({
+    templateUrl: '/source-report.component.html',
+    styleUrls: [ '/source-report.component.css' ]
+})
+export class SourceReportComponent implements OnInit, AfterViewChecked {
+    private execID: number;
+    private fileID: number;
+    private fileSource: string = "Loading...";
+    private fileLines: string[];
+
+    private fileModel: FileModel;
+    private sourceFileModel: SourceFileModel;
+    private transformedLinks: LinkModel[];
+
+    private classifications: ClassificationModel[];
+    private classificationLinks: Map<ClassificationModel, LinkModel[]> = new Map<ClassificationModel, LinkModel[]>();
+    private hints: InlineHintModel[];
+    private rendered: boolean = false;
+
+    constructor(private route:ActivatedRoute,
+                private fileModelService:FileModelService,
+                private classificationService:ClassificationService,
+                private hintService:HintService,
+                private http:Http
+    ) { }
+
+    ngOnInit(): void {
+        this.route.params.forEach((params: Params) => {
+            this.execID = +params['executionId'];
+            this.fileID = +params['fileId'];
+
```

Comment:
<s>I think you will have a problem here when you access this component from the nested route.</s>

fileId parameter should be available always, but for the route nested in migration-issues I'm not sure if you will see executionId. Hierarchical inheritance of route data and parameters is IMHO broken in angular and it **sometimes** inherits and sometimes not. 


Update:
After the second look on routing, I think it should inherit, because none of parent routes between report route and  source route has component or parameters defined.